packages/react/src/transactions/useCreatePost.ts

```diff
@@ -0,0 +1,354 @@
+ *       data: '0x.....'
+ *     }
+ *   ]
+ * });
+ * ```
+ *
+ * See {@link UnknownOpenActionConfig} for more details.
+ *
+ * ## Reference policy
+ *
+ * Contextually to the post creation you can configure the reference policy.
+ *
+ * A post with reference policy other than `ANYONE` will be hosted on-chain.
+ * If the post has reference policy `ANYONE` (which is also the default value) and does not have
+ * any open actions, it will be hosted on Momoka.
+ *
+ * No one can comment, quote, or mirror the post:
+ * ```tsx
+ * const result = await execute({
+ *   metadata: uri,
+ *
+ *   reference: {
+ *     type: ReferencePolicyType.NO_ONE
+ *   }
+ * });
+ * ```
+ *
+ * Only followers can comment, quote, or mirror the post:
+ * ```tsx
+ * const result = await execute({
+ *   metadata: uri,
+ *
+ *   reference: {
+ *     type: ReferencePolicyType.FOLLOWERS_ONLY
+ *   }
+ * });
+ * ```
+ *
+ * You can have finer control over who can comment, quote, or mirror the post by using the `DEGREES_OF_SEPARATION` reference policy:
+ * ```tsx
+ * const result = await execute({
+ *   metadata: uri,
+ *
+ *   reference: {
+ *     type: ReferencePolicyType.DEGREES_OF_SEPARATION,
+ *     params: {
+ *       degreesOfSeparation: 2, // followers and followers of your followers
+ *       commentsRestricted: true, // can comment
+ *       mirrorsRestricted: true, // can mirror
+ *       quotesRestricted: false, // cannot quote
+ *     }
+ *   }
+ * });
+ * ```
+ *
+ * You can even set the `DEGREES_OF_SEPARATION` reference policy to follow someone elses graph:
+ * ```tsx
+ * const result = await execute({
+ *   metadata: uri,
+ *
+ *   reference: {
+ *     type: ReferencePolicyType.DEGREES_OF_SEPARATION,
+ *     params: {
+ *       degreesOfSeparation: 2, // followers and followers of your followers
+ *       commentsRestricted: true, // can comment
+ *       mirrorsRestricted: true, // can mirror
+ *       quotesRestricted: false, // cannot quote
+ *
+ *       sourceProfileId: '0x01', // in relation to Profile Id 0x01
+ *     }
+ *   }
+ * });
+ * ```
+ *
+ * See {@link DegreesOfSeparationReferencePolicyConfig} for more details.
+ *
+ * ## Upgrading from v1
+ *
+ * Replace the `useCreatePost` hook with `useCreatePost` like in the following diff:
+ * ```diff
+ * - const { execute, error, isPending } = useCreatePost({ publisher, upload: uploadToIpfs });
+ * + const { execute, error, loading } = useCreatePost();
+ * ```
+ * Amend the code that used to call the `execute` function to:
+ * ```ts
+ * // first: create metadata using the `@lens-protocol/metadata` package
+ * const metadata = textOnly({
+ *   content: `Hello world!`,
+ * });
+ *
+ * // second: upload it using the upload function you used to pass to `useCreatePost`:
+ * const uri = await uploadToIpfs(metadata);
+ *
+ * // finally, invoke the `execute` function:
+ * const result = await execute({
+ *   metadata: uri,
+ * })
+ *
+ * // continue as usual
```

Comment:
this is all super cool, but how is this maintainable? 
wouldn't it be easier to put long form docs like this one in a markdown file? 
I don't know if that would work with typedoc, but we can output typedoc to markdown and combine with long-form docs in markdown and render them as webpages (maybe with docusaurus).
just something to think about for the future.

Reply:
I think we need to write it regardless. Checking in with the code is the first step. Cross link to other resources is also important.

TSDoc thicks both boxes.

I do agree, we should not exclude a mixed approach with MD based tools and TSDoc. For now we need to move, so this is better than nothing.